Source/WebCore/history/BackForwardCache.cpp

```diff
@@ -576,4 +577,19 @@ void BackForwardCache::prune(PruningReason pruningReason)
     }
 }
 
+void BackForwardCache::clearEntriesForOrigins(const HashSet<RefPtr<SecurityOrigin>>& origins)
+{
+    for (auto it = m_items.begin(); it != m_items.end();) {
+        // Increment iterator first so it stays valid after the removal.
+        auto current = it;
+        ++it;
+        auto itemOrigin = SecurityOrigin::create((*current)->url());
+        if (origins.contains(itemOrigin.ptr())) {
+            RELEASE_LOG(BackForwardCache, "BackForwardCache::clearEntriesForOrigins removing item: %s, size: %u / %u", (*current)->identifier().string().utf8().data(), pageCount() - 1, maxSize());
+            (*current)->setCachedPage(nullptr);
+            m_items.remove(current);
```

Comment:
Does this invalidate the iterator? Do we need to iterate in reverse, or accumulate the set of items to remove rather than removing while looping?

Reply:
The way we're looping and update `it` before doing the modification is safe, even if the container gets modified. The container in question is a ListHashSet and this is how we do safe traversal of a ListHashSet, even if the set gets modified. It is a ListHashSet feature. You'll also notice that this is how `BackForwardCache::removeAllItemsForPage(Page&)` is implemented in the same file.

Reply:
Obviously fine for now. Maybe we need a new alternate idiom for writing `ListHashSet` iteration and modification that doesn’t look like wrongly-written general purpose container iteration.